WordPress/AbstractThemeSniff.php

```diff
@@ -0,0 +1,1352 @@
+<?php
+/**
+ * Represents a PHP_CodeSniffer sniff for sniffing WordPress coding standards.
+ *
+ * @category  PHP
+ * @package   PHP_CodeSniffer
+ */
+
+/**
+ * Represents a PHP_CodeSniffer sniff for sniffing WordPress coding standards.
+ *
+ * Provides a bootstrap for the sniffs, to reduce code duplication.
+ *
+ * @category  PHP
+ * @package   PHP_CodeSniffer
+ * @author    Juliette Reinders Folmer <wpplugins_nospam@adviesenzo.nl>
+ * @author    khacoder
+ * @author    Ulrich Pogson <ulrich@pogson.ch>
+ */
+abstract class WordPress_AbstractThemeSniff implements PHP_CodeSniffer_Sniff {
+
+	/**
+	 * Array holding the once through WordPress Theme file checks.
+	 *
+	 * @var array
+	 */
+	private static $sniff_helper;
+
+	/**
+	 * Initial setup.
+	 */
+	public function __construct() {
+		// do a once through pass and get the files.
+		if ( empty( self::$sniff_helper ) || ! is_array( self::$sniff_helper ) || false === self::$sniff_helper ) {
+			$files = $this->get_files();
+			if ( isset( $files[0] ) ) {
```

Comment:
I will look at this when I get back. I have the complete sniff (my working files) now set up on the Theme check plugin. I am accessing the theme directory there and passing it to the AbstractSniff class in a using a.....wait for it.....global variable. It is working very well. The issue then becomes running the WordPress theme sniffs on a separate file, which the Abstract_ThemeSniff class is not intended for anyways.